tests/shell/shell_recorder/shell_recorder.sh

```diff
@@ -1,12 +1,24 @@
 #!/bin/bash
 
+getTempDir()
+{
+	if [ -z "$TMPDIR" ]; then
+		TMPDIR="/tmp"
+	else
+		if [[ "$TMPDIR" == */ ]]; then
+			TMPDIR=$(echo $TMPDIR | head -c -2)
+		fi
+	fi
+	echo "$TMPDIR"
+}
+
 FILE=$1
 Mountpoint=
 DBFile=
 Storage=
 MountArgs=
 DiffType=File
-OutFile=$(mktemp "${TMPDIR:-/tmp}/tmp.XXXXXXXXX")
+OutFile=$(mktemp "$(getTempDir)/tmp.XXXXXXXXX")
```

Comment:
You can leave it as is, but in the long term it might be a good idea to use include_common also for shell_recorder.

It has a mktempdir_elektra and mktempfile_elektra which does what you want here.